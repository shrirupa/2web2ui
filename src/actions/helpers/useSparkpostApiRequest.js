import { useState, useEffect } from 'react';
import { sparkpost as sparkpostAxios } from 'src/helpers/axiosInstances';
import store from 'src/store';
import _ from 'lodash';

function useSparkpostApiRequest(options) {
  const [response, setResponse] = useState({});

  // Force effect to rerun by using a toggled boolean as a dependant
  const [retryToggle, setRetryToggle] = useState(false);
  const retry = () => setRetryToggle(!retryToggle);

  const { auth } = store.getState();
  const httpOptions = { ...options };

  if (auth.loggedIn) {
    _.set(httpOptions, 'headers.Authorization', auth.token);
  }

  useEffect(() => {
    setResponse({ pending: true, error: null });

    sparkpostAxios(httpOptions).then((res) => {
      const results = _.get(res, 'data.results', res.data);
      setResponse({ pending: false, results });
    }).catch((error) => {
      setResponse({ pending: false, error: error });
    });
  }, [retryToggle]);

  return { ...response, retry };
}

export default useSparkpostApiRequest;
