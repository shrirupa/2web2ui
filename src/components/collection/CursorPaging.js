import styles from './CursorPaging.module.scss';
import { FirstPage } from '@sparkpost/matchbox-icons';
import React from 'react';
import { Button, Pager } from '@sparkpost/matchbox';

export const CursorPaging = ({ currentPage, handlePageChange, previousDisabled, nextDisabled, handleFirstPage, perPage, totalCount }) => (
  //previousDisabled and nextDisabled are boolean values.

  <div className={styles.PageButtons}>
    <Button className = {styles.RewindButton}
      key='rewind'
      onClick={handleFirstPage}
    >
      <FirstPage/>
    </Button>
    <Pager>
      <Pager.Previous
        onClick={() => handlePageChange(currentPage - 1)}
        disabled={previousDisabled}/>
      <Pager.Next
        onClick={() => handlePageChange(currentPage + 1)}
        disabled = {nextDisabled}/>
    </Pager>
    <span className={styles.PageDisplay}
    >Page: <strong className={styles.Bold}>{currentPage}</strong> / {Math.ceil(totalCount / perPage)}</span>
  </div>
);
export default CursorPaging;
