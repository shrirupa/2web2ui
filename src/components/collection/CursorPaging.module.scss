@import '~@sparkpost/matchbox/src/styles/config.scss';


.PageButtons {
  margin-bottom: spacing();
  @media screen and (min-width: breakpoint(small)) {
    width: 50%;
    float: left;
  }
}

.RewindButton {
  margin-right: spacing(small)*2;
}

.PageDisplay {
  font-size: 80%;
  text-transform: uppercase;
  color: rgba(0,0,0,0.5);
  margin-left: 20px;
}

.Bold {
  color: black;
  font-weight: bolder;
}
