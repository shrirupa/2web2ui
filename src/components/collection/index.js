export { default as Collection } from './Collection';
export { default as TableCollection } from './TableCollection';
export { default as FilterBox } from './FilterBox';
export { default as TableHeader } from './TableHeader';
export { default as NameCell } from './NameCell';
export { default as CursorPaging } from './CursorPaging';
export { default as PerPageButtons } from './PerPageButtons';
