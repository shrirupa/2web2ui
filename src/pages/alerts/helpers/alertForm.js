import {
  FILTERS_FRIENDLY_NAMES,
  METRICS,
  OPERATOR_FRIENDLY_NAMES,
  REALTIME_FILTERS,
  SIGNALS_FILTERS,
  SOURCE_FRIENDLY_NAMES,
  RECOMMENDED_METRIC_VALUE,
} from '../constants/formConstants';

export const getOptionsFromMap = (items, friendlyNameMap) =>
  items.map(item => ({ label: friendlyNameMap[item], value: item }));

const realtimeMetricsSpec = metric => ({
  hasFilters: true,
  filterType: 'multi',
  filterOptions: getOptionsFromMap(REALTIME_FILTERS, FILTERS_FRIENDLY_NAMES),
  sourceOptions: getOptionsFromMap(['raw'], SOURCE_FRIENDLY_NAMES),
  defaultRecommendedValue: RECOMMENDED_METRIC_VALUE[metric].raw.gt,
  defaultFieldValues: [
    { fieldName: 'source', fieldValue: 'raw' },
    { fieldName: 'operator', fieldValue: 'gt' },
  ],
});

const metricToFormSpecMap = {
  monthly_sending_limit: {
    hasFilters: false,
    filterType: null,
    filterOptions: [],
    sourceOptions: getOptionsFromMap(['raw'], SOURCE_FRIENDLY_NAMES),
    defaultRecommendedValue: RECOMMENDED_METRIC_VALUE['monthly_sending_limit'].raw.gt,
    defaultFieldValues: [
      { fieldName: 'subaccounts', fieldValue: [] },
      { fieldName: 'source', fieldValue: 'raw' },
      { fieldName: 'operator', fieldValue: 'gt' },
    ],
  },
  block_bounce_rate: realtimeMetricsSpec('block_bounce_rate'),
  hard_bounce_rate: realtimeMetricsSpec('hard_bounce_rate'),
  soft_bounce_rate: realtimeMetricsSpec('soft_bounce_rate'),
  health_score: {
    hasFilters: true,
    filterType: 'single',
    filterOptions: getOptionsFromMap(SIGNALS_FILTERS, FILTERS_FRIENDLY_NAMES),
    sourceOptions: getOptionsFromMap(
      ['raw', 'week_over_week', 'day_over_day'],
      SOURCE_FRIENDLY_NAMES,
    ),
    defaultRecommendedValue: RECOMMENDED_METRIC_VALUE['health_score'].raw.lt,
    defaultFieldValues: [
      { fieldName: 'source', fieldValue: 'raw' },
      { fieldName: 'operator', fieldValue: 'lt' },
    ],
  },
  injection_count: {
    hasFilters: false,
    filterType: null,
    filterOptions: [],
    sourceOptions: getOptionsFromMap(['raw'], SOURCE_FRIENDLY_NAMES),
    defaultFieldValues: [
      { fieldName: 'subaccounts', fieldValue: [] },
      { fieldName: 'source', fieldValue: 'raw' },
      { fieldName: 'operator', fieldValue: 'lt' },
      { fieldName: 'value', fieldValue: 100000 },
    ],
  },
  blacklist: {
    hasFilters: true,
    hideEvaluator: true,
    hideSubaccountFilter: true,
    filterType: 'multi',
    filterOptions: getOptionsFromMap(
      ['blacklist_provider', 'blacklist_resource'],
      FILTERS_FRIENDLY_NAMES,
    ),
    sourceOptions: [],
    defaultFieldValues: [{ fieldName: 'subaccounts', fieldValue: [] }],
  },
};

export const getFormSpec = metric => {
  if (metric in METRICS) {
    return metricToFormSpecMap[metric];
  }
  return {};
};

export const getEvaluatorOptions = (metric, source) => {
  const operatorArray =
    metric === 'injection_count'
      ? ['lt']
      : metric === 'health_score' && source === 'raw'
      ? ['lt', 'gt']
      : ['gt'];
  const operatorOptions = getOptionsFromMap(operatorArray, OPERATOR_FRIENDLY_NAMES);

  const suffix =
    ['health_score', 'injection_count'].includes(metric) && source === 'raw' ? '' : '%';

  const getSliderProps = () => {
    const sourceLabels = ['week_over_week', 'day_over_day'];
    if (sourceLabels.includes(source)) {
      return { label: 'Percent Change', precision: 0 };
    } else if (source !== 'raw') {
      return { label: 'Percent Change', precision: 2 };
    } else {
      switch (metric) {
        case 'health_score':
          return { label: 'Score', precision: 1 };
        case 'monthly_sending_limit':
          return { label: 'Percent Used', precision: 0 };
        case 'block_bounce_rate':
        case 'hard_bounce_rate':
        case 'soft_bounce_rate':
          return { label: 'Bounce Percentage Above', precision: 0 };
        case 'injection_count':
          return {
            label: 'Falls Below',
            max: 1000000,
            min: 1000,
            precision: 0,
          };
      }
    }

    return {};
  };

  return { suffix, operatorOptions, sliderProps: getSliderProps() };
};
