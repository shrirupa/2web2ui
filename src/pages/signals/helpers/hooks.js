// JSON.stringify() does not guarantee the serialised order of object members.
export const comparableJson = (obj) => JSON.stringify(Object.keys(obj).sort().map((key) => ({ [key]: obj[key] })));
