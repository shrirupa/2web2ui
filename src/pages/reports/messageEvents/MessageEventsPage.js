/* eslint-disable max-lines */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { snakeToFriendly } from 'src/helpers/string';
import { Button, Page } from '@sparkpost/matchbox';
import { PanelLoading, TableCollection, CursorPaging, PerPageButtons, ApiErrorBanner, Empty } from 'src/components';
import DisplayDate from 'src/components/displayDate/DisplayDate';
import MessageEventsSearch from './components/MessageEventsSearch';
import ViewDetailsButton from './components/ViewDetailsButton';
import { getMessageEvents, changePage, getMessageEventsCSV, clearCSV } from 'src/actions/messageEvents';
import { selectMessageEvents, selectMessageEventsCSV } from 'src/selectors/messageEvents';
import { formatToCsv, download } from 'src/helpers/downloading';
import { DEFAULT_PER_PAGE_BUTTONS } from 'src/constants';
import _ from 'lodash';
import styles from './MessageEventsPage.module.scss';

const errorMsg = 'Sorry, we seem to have had some trouble loading your message events.';
const emptyMessage = 'There are no message events for your current query';

const columns = [
  { label: 'Time' },
  { label: 'Event' },
  { label: 'Recipient' },
  { label: 'From Address' },
  null
];

export class MessageEventsPage extends Component {

  state = {
    currentPage: 1,
    perPage: 25
  }

  componentDidUpdate(prevProps) {
    const { search, getMessageEvents, eventsCSV } = this.props;
    const { perPage } = this.state;
    //Refresh the page & load new data if the search filters have changed
    if (!_.isEqual(prevProps.search, search)) {
      this.setState({ currentPage: 1 });
      getMessageEvents({ perPage, ...search });
    }
    if (!prevProps.eventsCSV.length && eventsCSV.length) {
      this.downloadCSV();
    }
  }

  downloadCSV = () => {
    const { eventsCSV, clearCSV } = this.props;
    const url = formatToCsv({ data: eventsCSV, returnBlob: true });
    const now = Math.floor(Date.now() / 1000);
    download({ name: `sparkpost-csv-${now}.csv`, url });
    clearCSV();
  };

  handlePageChange = (currentPage) => {
    this.setState({ currentPage });
    const { changePage } = this.props;
    return changePage(currentPage);
  }

  handlePerPageChange = (perPage) => {
    const { search, getMessageEvents } = this.props;
    this.setState({ perPage, currentPage: 1 });
    getMessageEvents({ perPage, ...search });
  }

  //Reload the first page w/ api call, NOT from cache
  handleFirstPage = () => {
    const { search, getMessageEvents } = this.props;
    const { perPage } = this.state;
    this.setState({ currentPage: 1 });
    getMessageEvents({ perPage, ...search });
  }

  isPreviousDisabled = () => {
    const { currentPage } = this.state;
    return currentPage <= 1;
  }

  isNextDisabled = () => {
    const { hasMorePagesAvailable } = this.props;
    return !hasMorePagesAvailable;
  }

  getCSV = () => {
    const { search, getMessageEventsCSV } = this.props;
    getMessageEventsCSV(search);
  }

  getRowData = (rowData) => {
    const { timestamp, formattedDate, type, friendly_from, rcpt_to } = rowData;
    return [
      <DisplayDate timestamp={timestamp} formattedDate={formattedDate} />,
      snakeToFriendly(type),
      rcpt_to,
      friendly_from,
      <ViewDetailsButton {...rowData} />
    ];
  }

  renderError() {
    const { error, getMessageEvents, search } = this.props;
    return (
      <ApiErrorBanner
        message={errorMsg}
        errorDetails={error.message}
        reload={() => getMessageEvents(search)}
      />
    );
  }

  renderCollection() {
    const { events, empty, loading, totalCount, eventsCSVLoading } = this.props;
    const { currentPage, perPage } = this.state;

    if (loading) {
      return <PanelLoading />;
    }

    const content = empty
      ? <Empty message={emptyMessage} />
      : (
        <div>
          <TableCollection
            columns={columns}
            rows={events}
            getRowData={this.getRowData}
            defaultSortColumn='timestamp'
            defaultSortDirection='desc'
          />
          <CursorPaging
            currentPage={currentPage}
            handlePageChange = {this.handlePageChange}
            previousDisabled={this.isPreviousDisabled()}
            nextDisabled={this.isNextDisabled()}
            handleFirstPage={this.handleFirstPage}
            perPage={perPage}
            totalCount={totalCount}
          />
          <div className={styles.RightAlignedButtons}>
            <PerPageButtons
              totalCount={totalCount}
              data={events}
              onPerPageChange={this.handlePerPageChange}
              perPageButtons={DEFAULT_PER_PAGE_BUTTONS}
              perPage={perPage}
              saveCsv={true}
            />
            <Button onClick={this.getCSV} disabled={eventsCSVLoading}>
              {(eventsCSVLoading) ? 'Saving CSV...' : 'Save as CSV'}
            </Button>
          </div>
        </div>
      );

    return content;
  }

  render() {
    const { error } = this.props;

    return (
      <Page title='Events Search'>
        <MessageEventsSearch />
        {error ? this.renderError() : this.renderCollection()}
      </Page>
    );
  }

}

const mapStateToProps = (state) => {
  const events = selectMessageEvents(state);
  const eventsCSV = selectMessageEventsCSV(state);
  const { messageEvents } = state;
  const { loading, error, search, totalCount, hasMorePagesAvailable, eventsCSVLoading } = messageEvents;
  return {
    events: events,
    loading,
    error,
    empty: events.length === 0,
    search,
    totalCount,
    hasMorePagesAvailable,
    eventsCSV,
    eventsCSVLoading
  };
};

export default connect(mapStateToProps, { getMessageEvents, getMessageEventsCSV, changePage, clearCSV })(MessageEventsPage);
