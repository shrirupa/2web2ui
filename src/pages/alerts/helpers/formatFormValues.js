import _ from 'lodash';
import { NOTIFICATION_CHANNELS } from '../constants/formConstants';
import { getFormSpec } from './alertForm';
import { multilineStringToArray } from 'src/helpers/string';

export default function formatFormValues(values) {
  const keysToOmit = [
    'value',
    'source',
    'operator',
    'sending_ip',
    'mailbox_provider',
    'blacklist_provider',
    'blacklist_resource',
    'sending_domain',
    'single_filter',
    ...NOTIFICATION_CHANNELS,
  ];
  const { metric, value, source, operator, single_filter, subaccounts } = values;

  const any_subaccount = subaccounts.length === 1 && subaccounts[0] === -2 ? true : undefined;

  //If subaccounts is empty, default to '-1'
  if (subaccounts.length === 0) {
    subaccounts.push(-1);
  }

  /*
  This gets the correct filters for the different metric types.
  First it looks up the form spec for that metric to get the form type.
  For single: returns empty array if no facet is selected, else return only that filter
  For multi: Loops through the realtime filters and adds the filters that have values.
  By default, return an empty array.
   */
  const { filterOptions, filterType, hideEvaluator } = getFormSpec(metric);

  const getFiltersMap = {
    single: () => (single_filter.filter_type === 'none' ? [] : [single_filter]),
    multi: () =>
      filterOptions
        .map(
          ({ value: filter_type }) =>
            values[filter_type].length > 0 && {
              filter_type,
              filter_values: values[filter_type],
            },
        )
        .filter(Boolean),
    default: () => [],
  };
  const filters = (getFiltersMap[filterType] || getFiltersMap.default)();

  const channels = {};
  const emails = (values.emails || '').trim();
  const slack = (values.slack || '').trim();
  const webhook = (values.webhook || '').trim();

  if (emails) {
    channels.emails = multilineStringToArray(emails);
  }

  if (slack) {
    channels.slack = { target: slack };
  }

  if (webhook) {
    channels.webhook = { target: webhook };
  }

  const keysToChange = {
    subaccounts: any_subaccount ? undefined : subaccounts,
    any_subaccount,
    filters,
    threshold_evaluator: hideEvaluator ? {} : { operator, source, value },
    channels,
  };

  return _.omit({ ...values, ...keysToChange }, keysToOmit);
}
