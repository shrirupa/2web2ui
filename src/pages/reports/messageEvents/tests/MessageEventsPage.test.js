import React from 'react';
import { shallow } from 'enzyme';
import { MessageEventsPage } from '../MessageEventsPage';
import { CursorPaging } from 'src/components/collection/CursorPaging';
import * as downloading from 'src/helpers/downloading';


let wrapper;
let instance;

describe('Page: Message Events tests', () => {
  const props = {
    empty: false,
    error: null,
    loading: false,
    getMessageEvents: jest.fn(() => []),
    getMessageEventsCSV: jest.fn(() => []),
    changePage: jest.fn(() => []),
    clearCSV: jest.fn(),
    refreshReportOptions: jest.fn(),
    events: [
      {
        formattedDate: 'so-formatted',
        type: 'delivery',
        friendly_from: 'hi@friendly',
        rcpt_to: 'ron.swanson@pawnee.state.in.us'
      },
      {
        formattedDate: 'formatted',
        type: 'injection',
        friendly_from: 'mean@friendly',
        rcpt_to: 'tom.haverford@pawnee.state.in.us'
      }
    ],
    history: {
      push: jest.fn()
    },
    search: {
      dateOptions: {},
      recipients: []
    },
    linkByPage: ['cursor=foo', 'cursor=bar', 'cursor=foobar', null],
    cachedResultsByPage: [[]],
    hasMorePagesAvailable: true,
    totalCount: 100,
    eventsCSV: [],
    eventsCSVLoading: false
  };

  beforeEach(() => {
    wrapper = shallow(<MessageEventsPage {...props} />);
    instance = wrapper.instance();
    wrapper.setState({
      currentPage: 2,
      perPage: 25
    });
    downloading.download = jest.fn();
    downloading.formatToCsv = jest.fn();
  });

  it('should render page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('refreshes on change in search filters', () => {
    const search = { ...props.search, changed: 'something' };
    wrapper.setProps({ search });
    expect(props.getMessageEvents).toHaveBeenCalledWith({ ...search, perPage: wrapper.state().perPage });
  });

  it('starts downloading events as CSV', () => {
    const eventsCSV = ['foo'];
    wrapper.setProps({ eventsCSV });
    expect(downloading.formatToCsv).toHaveBeenCalled();
    expect(downloading.download).toHaveBeenCalled();
  });

  it('does not refresh if search has not changed', () => {
    wrapper.setProps({ search: props.search, changed: 'something else' });
    expect(props.getMessageEvents).not.toHaveBeenCalled();
  });

  it('should render error when action fails', () => {
    wrapper.setProps({ error: { message: 'You done f\'ed up now' }});
    expect(wrapper).toMatchSnapshot();
    wrapper.find('ApiErrorBanner').props().reload();
    expect(wrapper.instance().props.getMessageEvents).toHaveBeenCalledWith({
      dateOptions: {},
      recipients: []
    });
  });

  it('should start retrieving CSV data when clicking Save as CSV', () => {
    wrapper.find('Button').last().simulate('click');
    expect(wrapper.instance().props.getMessageEventsCSV).toHaveBeenCalledWith({
      dateOptions: {},
      recipients: []
    });
  });

  it('should correctly disable previous button', () => {
    wrapper.setState({ currentPage: 1 });
    expect(wrapper.find(CursorPaging).prop('previousDisabled')).toEqual(true);
  });

  it('should correctly disable next button', () => {
    wrapper.setProps({ hasMorePagesAvailable: false });
    expect(wrapper.find(CursorPaging).prop('nextDisabled')).toEqual(true);
  });

  it('should correctly handle changing per page', () => {
    const perPage = 100;
    const { getMessageEvents, search } = props;
    instance.handlePerPageChange(perPage);
    expect(getMessageEvents).toHaveBeenCalledWith({ ...search, perPage });
  });

  it('should correctly handle changing page', () => {
    const { changePage } = props;
    const pageNumber = 1;
    instance.handlePageChange(pageNumber);
    expect(changePage).toHaveBeenCalledWith(pageNumber);
  });

  it('should correctly handle first page reload', () => {
    const { getMessageEvents, search } = props;
    instance.handleFirstPage();
    const { perPage } = wrapper.state();
    expect(getMessageEvents).toHaveBeenCalledWith({ ...search, perPage });
  });

  it('should only render loading component while loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the empty message when 0 results are returned', () => {
    wrapper.setProps({ empty: true });
    expect(wrapper).toMatchSnapshot();
  });

  describe('getRowData', () => {
    let event;
    beforeEach(() => {
      event = {
        formattedDate: 'formatted',
        type: 'injection',
        friendly_from: 'mean@friendly',
        rcpt_to: 'tom.haverford@pawnee.state.in.us',
        message_id: '123abc',
        event_id: '456xyz'
      };
    });

    it('renders correctly', () => {
      expect(instance.getRowData(event)).toMatchSnapshot();
    });

  });
});
