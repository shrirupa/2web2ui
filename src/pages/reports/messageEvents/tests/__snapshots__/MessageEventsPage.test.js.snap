// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Message Events tests should only render loading component while loading 1`] = `
<Page
  empty={Object {}}
  title="Events Search"
>
  <withRouter(Connect(MessageEventsSearch)) />
  <PanelLoading
    minHeight="400px"
  />
</Page>
`;

exports[`Page: Message Events tests should render error when action fails 1`] = `
<Page
  empty={Object {}}
  title="Events Search"
>
  <withRouter(Connect(MessageEventsSearch)) />
  <ApiErrorBanner
    errorDetails="You done f'ed up now"
    message="Sorry, we seem to have had some trouble loading your message events."
    reload={[Function]}
  />
</Page>
`;

exports[`Page: Message Events tests should render page correctly 1`] = `
<Page
  empty={Object {}}
  title="Events Search"
>
  <withRouter(Connect(MessageEventsSearch)) />
  <div>
    <TableCollection
      columns={
        Array [
          Object {
            "label": "Event",
          },
          Object {
            "label": "Subject",
          },
          Object {
            "label": "Recipient",
          },
          Object {
            "label": "From",
          },
          Object {
            "label": "Timestamp",
          },
        ]
      }
      defaultSortColumn="timestamp"
      defaultSortDirection="desc"
      rowComponent={[Function]}
      rows={
        Array [
          Object {
            "formattedDate": "so-formatted",
            "friendly_from": "hi@friendly",
            "rcpt_to": "ron.swanson@pawnee.state.in.us",
            "type": "delivery",
          },
          Object {
            "formattedDate": "formatted",
            "friendly_from": "mean@friendly",
            "rcpt_to": "tom.haverford@pawnee.state.in.us",
            "type": "injection",
          },
        ]
      }
    />
    <CursorPaging
      currentPage={2}
      handleFirstPage={[Function]}
      handlePageChange={[Function]}
      nextDisabled={false}
      perPage={25}
      previousDisabled={false}
      totalCount={100}
    />
    <div
      className="RightAlignedButtons"
    >
      <PerPageButtons
        data={
          Array [
            Object {
              "formattedDate": "so-formatted",
              "friendly_from": "hi@friendly",
              "rcpt_to": "ron.swanson@pawnee.state.in.us",
              "type": "delivery",
            },
            Object {
              "formattedDate": "formatted",
              "friendly_from": "mean@friendly",
              "rcpt_to": "tom.haverford@pawnee.state.in.us",
              "type": "injection",
            },
          ]
        }
        onPerPageChange={[Function]}
        perPage={25}
        perPageButtons={
          Array [
            10,
            25,
            50,
            100,
          ]
        }
        saveCsv={true}
        totalCount={100}
      />
      <Button
        disabled={false}
        onClick={[Function]}
        size="default"
      >
        Save as CSV
      </Button>
    </div>
  </div>
</Page>
`;

exports[`Page: Message Events tests should render the empty message when 0 results are returned 1`] = `
<Page
  empty={Object {}}
  title="Events Search"
>
  <withRouter(Connect(MessageEventsSearch)) />
  <Empty
    message="There are no message events for your current query"
  />
</Page>
`;
